### PR TITLE
Fix tx simulation for chains based on Cosmos SDK 0.42.x

### PR DESCRIPTION
**Pull request by romac**
Closes: #1354

## Description

In Cosmos SDK 0.43+, the Protocol Buffers message for `SimulateRequest` was updated, now included a `tx_bytes` field and deprecating the `tx` field:

```diff
pub struct SimulateRequest {
     /// tx is the transaction to simulate.
+    /// Deprecated. Send raw tx bytes instead.
+    #[deprecated]
     #[prost(message, optional, tag = "1")]
     pub tx: ::core::option::Option<Tx>,
+    /// tx_bytes is the raw transaction.
+    #[prost(bytes = "vec", tag = "2")]
+    pub tx_bytes: ::prost::alloc::vec::Vec<u8>,
 }
```

https://github.com/informalsystems/ibc-rs/commit/c4b6e3035c3b717f9f8ca09ec97c15beaf3d7cd9#diff-9ea64978cd0695e7d1321fc30fabcaa46a5857c514b638a22f6bad5f9ca25870R240-R248

As part of #948, we have changed the logic for building the `SimulateRequest` message accordingly, but did not realize that chains based on Cosmos SDK <0.43 would now reject the message with `invalid empty tx: invalid request`.

This would cause the gRPC call to fail and Hermes to then use the `max_gas` specified in the config to compute the fee for sending the actual transaction.

This PR fixes this by including both the `tx` and the `tx_bytes` fields so that chains requiring either field can process the request:

```rust
        let req = SimulateRequest {
             tx: Some(tx), // needed for simulation to go through with Cosmos SDK <  0.43
             tx_bytes,     // needed for simulation to go through with Cosmos SDk >= 0.43
         };
```

https://github.com/informalsystems/ibc-rs/pull/1405/files#diff-c12c5f52a7e35d9e3d8220ad3701dbe0f33dc1c21c7bf44518c44a5864f46907R406-R409

We may have to revisit this if the deprecated `tx` field is eventually removed, but this is outside the scope of this PR.

______

For contributor use:

- [x] Added a changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] If applicable: Unit tests written, added test to CI.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by Galadrin** [comment]
tested in production environment

https://www.mintscan.io/osmosis/txs/65E1A765970312E10A35F3DAE69A26B9F7BE8BB60455BA5A479D5510CE5F1344
https://www.mintscan.io/cosmos/txs/DB46FF1B3A85A2A6739C95AF376CD2F3516F1696A3526C232B0E766FC8F9DF6F

**Comment by romac** [comment]
Seems like the gas was estimated correctly there, taking into account the gas adjustment parameter.

So, would you say this fixed the issue you were seeing? If so, we can release this fix as part of Hermes 0.7.3 in the next couple days.

**Comment by Sdpierret** [comment]
I will keep it running during the next 24h

**Comment by Galadrin** [comment]
I will keepit running during the next 24h

**Review by soareschen — APPROVED** [review]

**Comment by mircea-c** [comment]
Just started running this in production (kek). Looks to be working as advertised. All our fees are now lower.

**Comment by ancazamfir** [comment]
@romac could you describe what the issue was exactly? Not clear to me after a first look at the diffs. 
Was the regression introduced  by #948?

File: relayer/src/chain/cosmos.rs [hunk 1]
```diff
@@ -379,12 +398,15 @@ impl CosmosSdkChain {
...
+        // The `tx` field of `SimulateRequest` was deprecated in Cosmos SDK 0.43 in favor of `tx_bytes`.
         let mut tx_bytes = vec![];
         prost::Message::encode(&tx, &mut tx_bytes).unwrap();
+
         #[allow(deprecated)]
-        let req = SimulateRequest { tx: None, tx_bytes };
+        let req = SimulateRequest {
+            tx: Some(tx), // needed for simulation to go through with Cosmos SDK <  0.43
+            tx_bytes,     // needed for simulation to go through with Cosmos SDk >= 0.43
+        };
```
Comment (ancazamfir) [inline relayer/src/chain/cosmos.rs:409]: ah, I guess this was the problem. Were you able to test the fix with both `< 0.43` and `>= 0.43`?

**Comment by romac** [comment]
> @romac could you describe what the issue was exactly? Not clear to me after a first look at the diffs.
> Was the regression introduced by #948?

Just updated the PR description.

File: relayer/src/chain/cosmos.rs (hunk 1 above)
Comment (romac) [reply relayer/src/chain/cosmos.rs:409]: Yes, I tested with both Gaia 5.0.5 and Gaia 6.0.0-rc1

**Pull request merged**